### PR TITLE
Notebook bug fixes and documentation updates 

### PR DESCRIPTION
**Pull request by axiomcura**
## Summary

This PR addresses a bug identified by one of the reviewers, as highlighted below:

> "_In the modeling notebook, the 'evaluating model on train dataset' code cell fails with the error: 'ValueError: The feature names should match those that were passed during fit. Feature names must be in the same order as they were in fit.' This issue may stem from changes in pandas affecting the order of selected features._"

In addition to fixing this bug, we have also updated the documentation to clarify the usage of the notebooks. The README now includes a "How to Use" section, which outlines the shell scripts associated with each module. Furthermore, we've added a section detailing the technologies utilized throughout the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +76,27 @@ The other is within the `./notebooks/4.visualization` directory, which includes
 
 That's it! Your Conda environments should now be set up with the specified packages from the YAML files.
 
+## How to use the notebook modules
+
+The notebooks are designed to be executed sequentially, with each module corresponding to a specific step in the process.
+Each module includes a `.sh` script that automates the execution of the notebooks within each module.
+
+All results generated by the notebooks are saved in the `./results` directory. This directory is organized with subfolders that are numbered according to the module from which the results were produced.
```
Comment (gwaybio) [inline README.md]:
```suggestion
All results generated by the notebooks are saved in the `./results` directory.
This directory is organized with subfolders that are numbered according to the module from which the results were produced.
```

one sentence per line

File: README.md [hunk 2]
```diff
@@ -118,6 +139,43 @@ We applied our trained multi-class logistic regression model to the JUMP dataset
...
+   cd ./notebooks/4.visualization/
+   ```
+
+2. Execute the bash script that sets up the Conda environment and runs the R scripts to generate the figures:
+
+   ```bash
+   source ./4.visualization.sh
+   ```
+
+This script will automatically create the required environment and execute the necessary R code. Once the process completes, all figures will be saved in the `./figures` directory.
```
Comment (gwaybio) [inline README.md]:
```suggestion
This script will automatically create the required environment and execute the necessary R code.
Once the process completes, all figures will be saved in the `./figures` directory.
```

File: README.md [hunk 3]
```diff
@@ -118,6 +139,43 @@ We applied our trained multi-class logistic regression model to the JUMP dataset
...
+
+## Development Tools Used
+
+In our project, we utilized a variety of development tools to enforce coding standards, formatting, and improve overall code quality. Below is a list of the primary technologies and linters used:
```
Comment (gwaybio) [inline README.md]:
```suggestion
In our project, we utilized a variety of development tools to enforce coding standards, formatting, and improve overall code quality. 
Below is a list of the primary technologies and linters used:
```

File: notebooks/1.data_splits/nbconverted/1.data_splits.py [hunk 4]
```diff
@@ -558,3 +566,9 @@ def get_and_rename_injury_info(
 
 # display
 merged_summary_df
+
+
+# In[20]:
+
+
+X_train
```
Comment (gwaybio) [inline notebooks/1.data_splits/nbconverted/1.data_splits.py]: typo to remove?

File: notebooks/2.modeling/nbconverted/2.modeling.py [hunk 5]
```diff
@@ -324,7 +324,7 @@
 )
 
 
-# In[14]:
+# In[ ]:
```
Comment (gwaybio) [inline notebooks/2.modeling/nbconverted/2.modeling.py]: perhaps you did not successfully nbconvert after saving the notebook?

**Review by gwaybio — APPROVED** [review]
LGTM @axiomcura - a couple minor comments to address prior to merging, no need for me to re-review.

What is super nice is for us to reference, in our response to reviewers, the specific pull requests that we performed to address specific reviewer comments. Consider adding reference to this pull request in the draft document (so that you don't have to do _all_ of them later). Do you understand this suggestion?

Also, along these lines (and for other best-practices reasons), this PR does 1 thing too much. You are performing "Notebook bug fixes **and** documentation updates**. Please don't get me wrong, the documentation updates are _top notch_, but, and especially for the purposes of a response to reviewers, try to keep pull requests to a single, focused changeset.

Excited to see the start of the response to reviewers! (great to see you add the specific comment to the PR description as well, keep it up!)

**Comment by axiomcura** [comment]
Hey Greg, Thank you for your feedback! All comments and suggestions have been addressed.

> _What is super nice is for us to reference, in our response to reviewers, the specific pull requests that we performed to address specific reviewer comments. Consider adding reference to this pull request in the draft document (so that you don't have to do all of them later). Do you understand this suggestion?_

Yes, I understand. I've included the link to this pull request in the "Response to Reviewers" document to track all the changes we've made.

> _Also, along these lines (and for other best-practices reasons), this PR does 1 thing too much. You are performing "Notebook bug fixes and documentation updates**. _

I agree! I'll be sure to separate these tasks in future pull requests. Thanks for the reminder!

Now merging!

**Pull request merged**